libs/RobotKit/include/RobotController.h

```diff
@@ -288,24 +288,40 @@ class RobotController : public interface::RobotController
 		raise(system::robot::sm::event::autonomous_activities_mode_requested {});
 	}
 
+	void raiseAutonomousActivityModeExited() { raise(system::robot::sm::event::autonomous_activities_mode_exited {}); }
+
 	void onMagicCardAvailable(const MagicCard &card)
 	{
+		using namespace std::chrono;
+
+		static auto start = rtos::Kernel::Clock::now();
+		static auto stop  = rtos::Kernel::Clock::now();
+
+		auto is_not_playing		= !_activitykit.isPlaying();
+		auto is_autonomous_mode = state_machine.is(system::robot::sm::state::autonomous_activities);
+
 		if (card == MagicCard::emergency_stop) {
 			raiseEmergencyStop();
 			return;
 		}
 
 		if (card == MagicCard::dice_roll) {
-			raiseAutonomousActivityModeRequested();
-			if (_activitykit.isPlaying()) {
-				_activitykit.stop();
+			if (is_autonomous_mode && is_not_playing) {
+				stop = rtos::Kernel::Clock::now();
+
+				if (stop - start > 2s) {
+					raiseAutonomousActivityModeExited();
+				}
+			} else {
+				start = rtos::Kernel::Clock::now();
+
+				if (start - stop > 1s) {
+					raiseAutonomousActivityModeRequested();
+				}
 			}
```

Comment:
as said on slack, I feel like the order of the checks does not reflect the order in which it happens in reality.

I would prefer something like this to show the different states/conditions in the order they appear:

1. not in autonomous mode
2. in autonomous mode && not playing
3. in autonomous mode && playing --> what happens in this case?

```cpp
if (is_not_autonomous_mode) {
	start = rtos::Kernel::Clock::now();

	if (start - stop > 1s) {
		raiseAutonomousActivityModeRequested();
	}

	return;
}

if (is_autonomous_mode && is_not_playing) {
	stop = rtos::Kernel::Clock::now();

	if (stop - start > 2s) {
		raiseAutonomousActivityModeExited();
	}

	return;
}

if (is_autonomous_mode && is_playing) {
	// do nothing
	return;
}
```

I personally don't really like the `else if` as I find it harder to read and to follow through the branching, but this case is simple enough to use it if you really want:

```cpp
if (is_not_autonomous_mode) {
	start = rtos::Kernel::Clock::now();

	if (start - stop > 1s) {
		raiseAutonomousActivityModeRequested();
	}
} else if (is_autonomous_mode && is_not_playing) {
	stop = rtos::Kernel::Clock::now();

	if (stop - start > 2s) {
		raiseAutonomousActivityModeExited();
	}
} else {
	// is_autonomous_mode && is_playing
	// do nothing
}
```